### ISSUE TITLE
setting wopi_url through occ config:app:set doesn't trigger discovery/enabling

### ISSUE THREAD

**Issue by robjuffermans** [nextcloud/richdocuments#1007]
When I set the wopi_url through `docker-compose exec --user www-data app php occ config:app:set richdocuments wopi_url --value=https://DOMAIN.TLD` the url under "URL (and Port) of Collabora Online-server" is set but there's still a red mark that says the server is unreachable.

However when I manually click the save button at the url the mark gets green and I can start using Collabora.

Does anyone had a simular experience?

Nextcloud domain and office-domain are both https. When I tail the query log of the database not much happens if I set the value through occ. Only an update of the wopi_url in the database. But when I click the save button much more is triggered.

Nextcloud 18.0.4
Collabora Online 3.6.0
setup via docker-compose

**Comment by juliushaertl** [comment]
Yes, that won't work the way the setup is currently implemented, since we fetch the discovery/capabilities endpoints of collabora to verify that the connection is working only when clicking the save button in the settings. There is no logic triggering this if the wopi_url is manually set. However it absolutely makes sense to have an occ command for that.

**Comment by robjuffermans** [comment]
Is there any way at this moment to automate this in a script so we end up with the green mark?

**Comment by t-markmann** [comment]
> 
> 
> Is there any way at this moment to automate this in a script so we end up with the green mark?
```

appdir=$(find /pathtoyour/data/ -maxdepth 1 -type d -name "appdata_*" -print)

sudo -u www-data php occ config:app:set richdocuments wopi_url --value=https:\/\/collabora.yourdomain.tld
sudo -u www-data php occ config:app:set richdocuments public_wopi_url --value=https:\/\/collabora.yourdomain.tld

sudo rm ${appdir}/richdocuments/richdocuments/discovery.xml

```
Of course you can just set the appdir manually.
The collabora app will create the discovery.xml again, when a document is loaded. Hope it helps!

**Comment by robjuffermans** [comment]
Thanks t-markmann, but I'm afraid as @juliushaertl pointed out that at this moment there's no way to script something to generate the discovery.xml and get a up&running collabora without pressing the save button in the collabora app settings.

**Comment by juliushaertl** [comment]
Contributions for an occ command that updates the capabilities and discovery properly are of course also very welcome :wink:

**Comment by ebardie** [comment]
@t-markmann : {"productName":"LibreOffice Online"} works on nc19.0.2 here

**Comment by ebardie** [comment]
I'm looking at creating a command called 'activate-config' to parallel the code triggered by clicking 'Save' in the WUI form.

**Comment by juliushaertl** [comment]
As a small hint for the code, https://github.com/nextcloud/richdocuments/blob/3e53a966c320548162d991ef43b72420b5cc4c72/lib/Controller/SettingsController.php#L143-L198 would be basically needed to be moved to a helper function and then it could be reused for both the ui saving as well as the occ command.

**Comment by ebardie** [comment]
Yes, that was my start point, but the WUI code has too many contextual assumptions. The occ command context knows nothing about what the config changes are, or even if there have been any, and there's no web request involved, so the protocol check, the early return, and message return don't make sense.

So I've extracted the minimum necessary code and submitted PR #1137.

I've tested the new command in my Ansible use case.
